src/python/genny/genny_auto_tasks.py

```diff
@@ -45,17 +80,83 @@ def modified_workload_files():
     return short_filenames
 
 
-def get_project_root():
+def workload_should_autorun(workload_dict, env_dict):
     """
-    :return: the path of the project root.
+    Check if the given workload's AutoRun conditions are met by the current environment
+    :param dict workload_dict: a dict representation of the workload files's yaml.
+    :param dict env_dict: a dict representing the values from bootstrap.yml and runtime.yml
+    :return: True if this workload should be autorun, else False.
     """
-    try:
-        out = subprocess.check_output('git rev-parse --show-toplevel', shell=True)
-    except subprocess.CalledProcessError as e:
-        print(e.output, file=sys.stderr)
-        raise e
 
-    return out.decode().strip()
+    # First check that the workload has a proper AutoRun section.
+    if 'AutoRun' not in workload_dict or not isinstance(workload_dict['AutoRun'], dict):
```

Comment:
Similar to my comment above, I think we should go the extra mile of making an object representation of the AutoRun section of the YAML file. This way we can avoid typing strings and make it easy to rename fields. Talk to @aggrand if you want examples :)